src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,249 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private Node<K, V>[] arrayNodes;
+    private final int defaultCapacity = 16;
+    private final float loadFactor = 0.75f;
+    private int threshold;
+    private int size;
+
+    public MyHashMap() {
+        this.arrayNodes = (Node<K, V>[]) new Node[defaultCapacity];
+        this.threshold = (int) (loadFactor * arrayNodes.length);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> newNode = new Node<>(key, value);
+        int backet = getHash(newNode);
+        if (backet < 0 || backet > arrayNodes.length) {
+            throw new IndexOutOfBoundsException();
+        }
+        if (arrayNodes[backet] == null) {
+            simplePut(newNode);
+            return;
+        }
+        if (getHash(arrayNodes[backet]) == backet) {
+            putWithCollision(newNode);
+        }
+        if (size == threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        V value;
+        if (key == null) {
+            return getValueByNullKey(key);
+        }
+        for (Node<K, V> nodeForGetValue : arrayNodes) {
```

Comment:
do we need to go through all nodes in the table, if we can calculate in what bucket we should looking for node with such key?